### PR TITLE
Update Developer Guide to include Implementation of Dynamic Text Field

### PR DESCRIPTION
**Pull request by reignnz**

### CODE REVIEW

File: docs/diagrams/DynamicInputFindDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+activate AddressBookParser
+
+create FindCommandParser
+AddressBookParser -> FindCommandParser
+activate FindCommandParser
+
+FindCommandParser -> AddressBookParser
+deactivate FindCommandParser
+
+AddressBookParser -> FindCommandParser : parse(“Alex)
```
Comment (zihaowrez) [inline docs/diagrams/DynamicInputFindDiagram.puml:31]: closing quote

**Review by zihaowrez — CHANGES_REQUESTED** [review]
Maybe need to explain how the input "Alex" becomes "dynamic Alex"?

File: docs/diagrams/DynamicInputFindDiagram.puml (hunk 1 above)
Comment (reignnz) [reply docs/diagrams/DynamicInputFindDiagram.puml]: Noted. Thank you

File: docs/diagrams/DynamicInputFindDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+box Logic LOGIC_COLOR_T1
+participant “:CommandBox” as CommandBox LOGIC_COLOR
+participant “:MainWindow” as MainWindow LOGIC_COLOR
+participant “:LogicManager” as LogicManager LOGIC_COLOR
+participant “:AddressBookParser” as AddressBookParser LOGIC_COLOR
+participant “:FindCommandParser” as FindCommandParser LOGIC_COLOR
+end box
+
+[-> CommandBox : handleDynamicInput()
```
Comment (zihaowrez) [inline docs/diagrams/DynamicInputFindDiagram.puml:12]: should this be `handleDynamicInput("Alex")`?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -234,6 +234,31 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+handleDynamicInput(), reads the user input at each deletion or addition of the command in the
+CLI and calls MainWindow#executeCommand. It passes the command inputted by the user with the
+string "dynamic" concatted to the front, and a reference of itself (a CommandBox object).
+
+The user input and instance of commandBox object is then passed to LogicManager#execute and
+subsequently AddressBookParser#parseCommand and FindCommand#parse(arguments).
+
+The above is assuming that the user inputs a string not included in the
+list of commands: “add”, “delete”, “list”, “find”, “view”, “edit”.
+
```
Comment (zihaowrez) [inline docs/DeveloperGuide.md:253]: Is "copy" also recognised?

**Review by zihaowrez — APPROVED** [review]
LGTM

**Pull request merged**